reframe/utility/sanity.py

```diff
@@ -547,6 +547,8 @@ def calc_bound(thres):
     except SanityError:
         error_msg = msg or '{0} is beyond reference value {1} (l={2}, u={3})'
         raise SanityError(_format(error_msg, val, ref, lower, upper))
+    except TypeError:
```

Comment:
The fix should go inside the actual comparison in the [assert_bounded()](https://github.com/eth-cscs/reframe/blob/d38da74943e3317ae1fc6f541e2aaf837307c2ef/reframe/utility/sanity.py#L492) function.